src/include/api.h

```diff
@@ -231,47 +233,47 @@
 #define SESSION_API_CALL_PREPARE_ALLOWED_NOCONF(s, func_name) \
     API_CALL_NOCONF(s, WT_SESSION, func_name, NULL)
 
-#define SESSION_API_CALL_PREPARE_NOT_ALLOWED(s, func_name, config, cfg) \
-    SESSION_API_PREPARE_CHECK(s, WT_SESSION, func_name);                \
-    API_CALL(s, WT_SESSION, func_name, NULL, config, cfg)
-
-#define SESSION_API_CALL_PREPARE_NOT_ALLOWED_NOCONF(s, func_name) \
-    SESSION_API_PREPARE_CHECK(s, WT_SESSION, func_name);          \
-    API_CALL_NOCONF(s, WT_SESSION, func_name, NULL)
-
-#define SESSION_API_PREPARE_CHECK(s, struct_name, func_name)   \
-    do {                                                       \
-        if ((s)->api_call_counter == 0) {                      \
-            int __prepare_ret;                                 \
-            API_SESSION_PUSH(s, WT_SESSION, func_name, NULL);  \
-            __prepare_ret = __wt_txn_context_prepare_check(s); \
-            API_SESSION_POP(s);                                \
-            WT_RET(__prepare_ret);                             \
-        }                                                      \
+#define SESSION_API_CALL_PREPARE_NOT_ALLOWED(s, ret, func_name, config, cfg) \
+    API_CALL(s, WT_SESSION, func_name, NULL, config, cfg);                   \
+    SESSION_API_PREPARE_CHECK(s, ret, WT_SESSION, func_name)
+
+#define SESSION_API_CALL_PREPARE_NOT_ALLOWED_NOCONF(s, ret, func_name) \
+    API_CALL_NOCONF(s, WT_SESSION, func_name, NULL);                   \
+    SESSION_API_PREPARE_CHECK(s, ret, WT_SESSION, func_name)
+
+#define SESSION_API_PREPARE_CHECK(s, ret, struct_name, func_name) \
+    do {                                                          \
+        if ((s)->api_call_counter == 1) {                         \
+            (ret) = __wt_txn_context_prepare_check(s);            \
+            if ((ret) != 0) {                                     \
+                __set_err = false;                                \
```

Comment:
Can we have a comment here explaining why we don't want to call `__wt_txn_err_set` when there's an error?

Reply:
done